### ISSUE TITLE
Test Data For Col Defects

### ISSUE THREAD

**Issue by jeffeastman** [DBCG/cqf-ruler#5]
Here's the test data from the remaining issues reported on thread #4  "Cross-Posting Issue From cal_engine" which is closed

[col_encounter.json.txt](https://github.com/DBCG/cqf-ruler/files/1168278/col_encounter.json.txt)

**Comment by jeffeastman** [comment]
Here's another file with a Colonoscopy instead of Sigmoidoscopy that has similar issues. I've got some pretty good tools for creating this sort of test data now. Let me know if you have any specific needs other than these measures.

My terminology service
```
ValueSet 2.16.840.1.113883.3.464.1003.108.11.1001 contains 'http://snomed.info/sct' 285312008
Adding resource - Condition-1
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 does not contain 'http://snomed.info/sct' 73761001
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 contains 'http://www.ama-assn.org/go/cpt' 44155
Adding resource - Procedure-14
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 contains 'http://snomed.info/sct' 73761001
Adding resource - Procedure-9
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://www.ama-assn.org/go/cpt' 44155
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 14563-1
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 contains 'http://www.ama-assn.org/go/cpt' 45378
Adding resource - Observation-10
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://snomed.info/sct' 73761001
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://www.ama-assn.org/go/cpt' 44155
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 14563-1
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://www.ama-assn.org/go/cpt' 45378
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 contains 'http://loinc.org' 2335-8
Adding resource - Procedure-4
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://snomed.info/sct' 73761001
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://www.ama-assn.org/go/cpt' 44155
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 contains 'http://loinc.org' 14563-1
Adding resource - Observation-5
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://www.ama-assn.org/go/cpt' 45378
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3
2017-07-23 14:19:23 INFO  FhirContext:172 - Creating new FHIR context for FHIR version [DSTU3]

```

cqf-ruler terminology service

```
ValueSet 2.16.840.1.113883.3.464.1003.108.11.1001 does not contain 'http://snomed.info/sct' 285312008
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 does not contain 'http://snomed.info/sct' 73761001
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 does not contain 'http://www.ama-assn.org/go/cpt' 44155
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://snomed.info/sct' 73761001
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://www.ama-assn.org/go/cpt' 44155
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 14563-1
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 contains 'http://www.ama-assn.org/go/cpt' 45378
Adding resource - Observation-10
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://snomed.info/sct' 73761001
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://www.ama-assn.org/go/cpt' 44155
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 14563-1
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://www.ama-assn.org/go/cpt' 45378
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 contains 'http://loinc.org' 2335-8
Adding resource - Procedure-4
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://snomed.info/sct' 73761001
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://www.ama-assn.org/go/cpt' 44155
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 contains 'http://loinc.org' 14563-1
Adding resource - Observation-5
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://www.ama-assn.org/go/cpt' 45378
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3

```

And the test bundle: 
[col_encounter2.json.txt](https://github.com/DBCG/cqf-ruler/files/1168305/col_encounter2.json.txt)

**Comment by jeffeastman** [comment]
Here's one that has all of the groups populated. Simpler than importing both of the above because of the id overloading.

[col_encounter_all.json.txt](https://github.com/DBCG/cqf-ruler/files/1168319/col_encounter_all.json.txt)

**Comment by c-schuler** [comment]
I populated the hapi server with the resources you provided (I edited them to reference Patient-12214 and Practitioner-12208 and removed the encounter and location references for simplicity) and ran the following: 

http://measure.eval.kanvix.com/cqf-ruler/baseDstu3/Measure/col/$evaluate?patient=Patient-12214&startPeriod=2016-01-01&endPeriod=2016-12-31

All the resources were returned in the report successfully. What period dates are you using to test?

**Comment by jeffeastman** [comment]
My period dates are consistent with the test data I’m using (today - 1 yr … today). The issue I’m seeing is with the ValueSet memberships and not with the period dates.

**Comment by c-schuler** [comment]
Is this still an issue? I was unable to reproduce with the resources you provided...

**Comment by jeffeastman** [comment]
Yes. When I run COL locally, I get this result:

```
Theodore Lonnie Villarreal [60]
ValueSet 2.16.840.1.113883.3.464.1003.108.11.1001 contains 'http://snomed.info/sct' 94105000
Adding resource - Condition-1
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 does not contain 'http://www.ama-assn.org/go/cpt' 45331
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1019 contains 'http://www.ama-assn.org/go/cpt' 44156
Adding resource - Procedure-74
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://www.ama-assn.org/go/cpt' 45331
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://www.ama-assn.org/go/cpt' 44156
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 12504-7
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://www.ama-assn.org/go/cpt' 45340
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.108.12.1020 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 2335-8
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 contains 'http://www.ama-assn.org/go/cpt' 45331
Adding resource - Procedure-49
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://www.ama-assn.org/go/cpt' 44156
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 12504-7
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 contains 'http://www.ama-assn.org/go/cpt' 45340
Adding resource - Observation-50
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1010 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 contains 'http://loinc.org' 2335-8
Adding resource - Procedure-8
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://www.ama-assn.org/go/cpt' 45331
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://www.ama-assn.org/go/cpt' 44156
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 contains 'http://loinc.org' 12504-7
Adding resource - Observation-9
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://www.ama-assn.org/go/cpt' 45340
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3
ValueSet 2.16.840.1.113883.3.464.1003.198.12.1011 does not contain 'http://loinc.org' 8716-3
	in-demographic	count= 1
	history-of-colorectal-cancer	count= 1
	history-of-total-colectomy	count= 1
	colonoscopy-performed	count= 0
	colonoscopy-results	count= 0
	sigmoidoscopy-procedure	count= 1
	sigmoidoscopy-observation	count= 1
	fobt-procedure	count= 1
	fobt-observation	count= 1

```

but when I run against cqf-ruler-ruler with the same patients, I get this:

```
Sending 'GET' request to URL: http://localhost:8080/cqf-ruler/baseDstu3/Measure/col/$evaluate?patient=Patient-2&startPeriod=2020-01&endPeriod=2020-12&_raw=true
200
http://localhost:8080/cqf-ruler/baseDstu3/Patient/Patient-2/_history/1
in-demographic	count= 1
history-of-colorectal-cancer	count= 1
history-of-total-colectomy	count= 1
colonoscopy-performed	count= 0
colonoscopy-results	count= 0
sigmoidoscopy-procedure	count= 1
sigmoidoscopy-observation	count= 1
fobt-procedure	count= 0
fobt-observation	count= 0

```
so it looks to me like the fobt value set `<def name="Fecal Occult Blood Test (FOBT)" id="2.16.840.1.113883.3.464.1003.198.12.1011" accessLevel="Public"/>` is not matching properly. Unfortunately, I cannot get the same checkCodeMembership debugging on the server so I'm stuck.

**Comment by c-schuler** [comment]
The period dates are incorrect as they do not correspond with the FOBT resources associated with the patient. Works just fine with 2016 dates:
http://measure.eval.kanvix.com/cqf-ruler/baseDstu3/Measure/col/$evaluate?patient=Patient-2&startPeriod=2016-01&endPeriod=2016-12

Note the "Lookback" date interval filter will exclude any resources not performed within the last year. That would be in the 2019-2020 range. However, the resource's performedPeriod and effectiveDateTime properties are for 2016. Therefore, the resources are excluded.
```
define "FOBT Procedure":
  [Procedure: "Fecal Occult Blood Test (FOBT)"] F
    where F.status.value = 'completed'
      and F.performedPeriod."end".value during "Lookback Interval One Year"

define "FOBT Observation":
  [Observation: "Fecal Occult Blood Test (FOBT)"] O
    where O.status.value = 'final'
      and O.effectiveDateTime.value during "Lookback Interval One Year"
```

I ran some other tests as well just to be sure. Everything seems to be in good working order.
Ran this, which returned Procedure 113 with FOBT code 2335-8:
http://measure.eval.kanvix.com/cqf-ruler/baseDstu3/Measure/col/$evaluate?patient=110&startPeriod=2020-01-01&endPeriod=2020-12-31

Also ran this, which returned Procedure 252 with FOBT code 12504-7:
http://measure.eval.kanvix.com/cqf-ruler/baseDstu3/Measure/col/$evaluate?patient=249&startPeriod=2020-01-01&endPeriod=2020-12-31
